internal/telemetrygen/common/config.go

```diff
@@ -0,0 +1,158 @@
+//
+// It modifies the original implementation to remove unnecessary files,
+// and to accept a configurable logger.
+
+package common
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/zap"
+)
+
+var (
+	errFormatOTLPAttributes       = fmt.Errorf("value should be of the format key=\"value\"")
+	errDoubleQuotesOTLPAttributes = fmt.Errorf("value should be a string wrapped in double quotes")
+)
+
+const (
+	defaultGRPCEndpoint = "localhost:4317"
+	defaultHTTPEndpoint = "localhost:4318"
+)
+
+type KeyValue map[string]string
+
+var _ pflag.Value = (*KeyValue)(nil)
+
+func (v *KeyValue) String() string {
+	return ""
+}
+
+func (v *KeyValue) Set(s string) error {
+	kv := strings.SplitN(s, "=", 2)
+	if len(kv) != 2 {
+		return errFormatOTLPAttributes
+	}
+	val := kv[1]
+	if len(val) < 2 || !strings.HasPrefix(val, "\"") || !strings.HasSuffix(val, "\"") {
+		return errDoubleQuotesOTLPAttributes
+	}
+
+	(*v)[kv[0]] = val[1 : len(val)-1]
+	return nil
+}
+
+func (v *KeyValue) Type() string {
+	return "map[string]string"
+}
+
+type Config struct {
+	WorkerCount           int
+	Rate                  int64
+	TotalDuration         time.Duration
+	ReportingInterval     time.Duration
+	SkipSettingGRPCLogger bool
+
+	// OTLP config
+	CustomEndpoint      string
+	Insecure            bool
+	UseHTTP             bool
+	HTTPPath            string
+	Headers             KeyValue
+	ResourceAttributes  KeyValue
+	TelemetryAttributes KeyValue
+
+	// OTLP TLS configuration
+	CaFile string
+
+	// OTLP mTLS configuration
+	ClientAuth ClientAuth
+
+	Logger *zap.Logger
+}
+
+type ClientAuth struct {
+	Enabled        bool
+	ClientCertFile string
+	ClientKeyFile  string
+}
+
+// Endpoint returns the appropriate endpoint URL based on the selected communication mode (gRPC or HTTP)
+// or custom endpoint provided in the configuration.
+func (c *Config) Endpoint() string {
+	if c.CustomEndpoint != "" {
+		return c.CustomEndpoint
+	}
+	if c.UseHTTP {
+		return defaultHTTPEndpoint
+	}
+	return defaultGRPCEndpoint
+}
+
+func (c *Config) GetAttributes() []attribute.KeyValue {
+	var attributes []attribute.KeyValue
+
+	if len(c.ResourceAttributes) > 0 {
+		for k, v := range c.ResourceAttributes {
+			attributes = append(attributes, attribute.String(k, v))
+		}
+	}
+	return attributes
+}
+
+func (c *Config) GetTelemetryAttributes() []attribute.KeyValue {
+	var attributes []attribute.KeyValue
+
+	if len(c.TelemetryAttributes) > 0 {
+		for k, v := range c.TelemetryAttributes {
+			attributes = append(attributes, attribute.String(k, v))
+		}
+	}
+	return attributes
+}
+
+// CommonFlags registers common config flags.
+func (c *Config) CommonFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&c.WorkerCount, "workers", 1, "Number of workers (goroutines) to run")
+	fs.Int64Var(&c.Rate, "rate", 0, "Approximately how many metrics per second each worker should generate. Zero means no throttling.")
```

Comment:
*suggestion*: Our loadgen also provides a similar feature ([ref](https://github.com/elastic/apm-perf/blob/c1b642c8603c46291329e8672585a2dd46dd0326/internal/loadgen/limiter.go#L13-L18)) but has a different interface, maybe in future we can update this to have a similar option OR even better use the same limiter code for both cases.